src/backend/api/meals.js

```diff
@@ -2,14 +2,72 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+//select all
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = await knex("meal").select("*");
+    response.json(meals);
+  } catch (error) {
+    throw error;
+  }
+});
+//new meal
+router.post("/", async (request, response) => {
+  const addmeal = request.body;
+  const [mealsId] = await knex("meal").insert({
+    title: addmeal.title,
+    description: addmeal.description,
+    location: addmeal.location,
+    when: addmeal.when,
+    max_reservations: addmeal.max_reservations,
+    price: addmeal.price,
+    created_date: addmeal.created_date,
+  });
+  if (mealsId === 0) {
+    response.status(404).json({ message: "No meals found" });
+  } else {
+    response.status(201).json({ message: "New created meal", id: mealsId });
+  }
+});
+// return the meal by id
+router.get("/:id", async (request, response) => {
+  try {
+    const mealSearchId = await knex("meal").where({
+      id: request.params.id,
+    });
+    if (mealSearchId.length > 0) {
+      response.json(mealSearchId);
+    } else {
+      response.status(404).send(`This Id in not found`);
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+// update the meal by id
+router.put("/:id", async (request, response) => {
+  try {
+    const mealUpdate = request.body;
+    const mealIdUpdate = await knex("meal")
+      .where({ id: request.params.id })
+      .update({
+        title: mealUpdate.title,
+        description: mealUpdate.description,
+      });
+    response.json(mealIdUpdate);
```

Comment:
This would just return a number, usually just 1. I think a JSON object like `{ message: "Meal was update" }` would be more descriptive.

Also, if we're trying to update a meal that does not exist, we should return a 404 response. I don't think your code does that here currently.